### ISSUE TITLE
FieldProps Flow Error redux-form@8.2.0

### ISSUE THREAD

**Issue by Solles** [erikras/redux-form#4434]
### Are you submitting a **bug report** or a **feature request**?
bug report

### What is the current behavior?
```react
<Field
	autoCapitalize="none"
	component={FormInput}
	keyboardType="email-address"
	label="Username"
	name="username"
/>
```
and FormInput's props are

```flow
type Props = {
	label: string,
} & FieldProps;
```

![image](https://user-images.githubusercontent.com/13891449/55966114-d191c180-5c45-11e9-8c17-602b20bfdece.png)

### What is the expected behavior?
I expect the above to not return a Flow error.

### Sandbox Link
https://codesandbox.io/s/gJjDoKYWl seems to demonstrate an example of using a Field to pass-through properties.

### What's your environment?
redux-form: 8.2.0
react-native:  0.95.4
react-redux: 7.0.1
flow: 0.96.0
node: 11.10.1

### Other information
This flow error doesn't occur in 8.1.0 but does occur in 8.2.0.

I think it's because it doesn't like that I'm trying to pass properties through to my underlying component. Is it related to the changes in [this commit](https://github.com/erikras/redux-form/commit/a6b626dd795b898683a10a9245bc7af86cfcd385)?

**Comment by montogeek** [comment]
Yes, it is definitely related to that change, I just got the same error to any additional prop that you pass.

**Comment by montogeek** [comment]
cc @asazernik

**Comment by montogeek** [comment]
By changing it to: `component: ComponentType<*> | string,` will fix all issues.

**Comment by asazernik** [comment]
Ah - yeah, that's a code style I'm not using and forgot to test for :-\

Will put together a PR right now to relax that prop-type constraint - a stricter-but-correct version will require some research.

**Comment by asazernik** [comment]
Alright, there's the PR. @erikras, hotfix?

**Comment by orlandodemauro** [comment]
@erikras @asazernik I´m facing the same issue. any suggestion or update on the status of this issue? thanks a milion

**Comment by orlandodemauro** [comment]
@renatoagds thank you for merging the pr. any idea on when it will be released? thanks

**Comment by madeleinemf** [comment]
Yes - would be good to have a release of this soon.

**Comment by renatoagds** [comment]
@orlandodemauro until the final of the week we'll have a release.

**Comment by erikras** [comment]
Published fix in [`v8.2.2`](https://github.com/erikras/redux-form/releases/tag/v8.2.2).

**Comment by orlandodemauro** [comment]
@renatoagds @erikras thanks guys
